evil-collection-buff-menu.el

```diff
@@ -27,10 +27,83 @@
 ;;; Bindings for `buff-menu'.
 
 ;;; Code:
+
+;; taken from emacs upstream repository with a minnor change
+;; this is all redundant code after emacs 26 
+(require 'tabulated-list) ;; required for evil-collection-Buffer-menu-unmark-all-buffers
+
+(defsubst evil-collection-tabulated-list-header-overlay-p (&optional pos)
+  "Return non-nil if there is a fake header.
+Optional arg POS is a buffer position where to look for a fake header;
+defaults to `point-min'."
+  (overlays-at (or pos (point-min))))
+
+(defun evil-collection-Buffer-menu-unmark-all ()
+  "Cancel all requested operations on buffers."
+  (interactive)
+  (evil-collection-Buffer-menu-unmark-all-buffers ?\r))
+
+(defun evil-collection-Buffer-menu-unmark-all-buffers (mark)
+  "Cancel a requested operation on all buffers.
+MARK is the character to flag the operation on the buffers.
+When called interactively prompt for MARK;  RET remove all marks."
+  (interactive "cRemove marks (RET means all):")
+  (save-excursion
+    (goto-char (point-min))
+    (when (evil-collection-tabulated-list-header-overlay-p)
+      (forward-line))
+    (while (not (eobp))
+      (let ((xmarks (list (aref (tabulated-list-get-entry) 0)
+			  (aref (tabulated-list-get-entry) 2))))
+	(when (or (char-equal mark ?\r)
+		  (member (char-to-string mark) xmarks))
+	  (Buffer-menu--unmark)))
+      (forward-line))))
+
 (defun evil-collection-buff-menu-setup ()
   "Set up `evil' bindings for `buff-menu'.."
-  ;; FIXME: Fill me in!
-  (evil-add-hjkl-bindings Buffer-menu-mode-map 'motion))
+
+  (evil-set-initial-state 'Buffer-menu-mode 'normal)
+  (evil-add-hjkl-bindings Buffer-menu-mode-map 'normal)
+
+  (evil-define-key 'normal Buffer-menu-mode-map
+    "ZQ" 'evil-quit
+    "ZZ" 'quit-window
+    "gr" 'revert-buffer
+    "go" 'Buffer-menu-this-window
+    "gO" 'Buffer-menu-other-window
+    "d"  'Buffer-menu-delete
+    "u"  'Buffer-menu-unmark 
+    "U"  'evil-collection-Buffer-menu-unmark-all ;; TODO: add this
```

Comment:
Does it make sense to do a version check and use the emacs-26 version if it's available?

Reply:
I checked and it is in the emacs-26 branch. I don't know how much time there is until emacs 26. I will do it anyways so that emacs-26 users don't load redundant code.